app/src/main/java/org/jellyfin/androidtv/browsing/HomeFragment.java

```diff
@@ -132,283 +150,223 @@ public void onResume() {
-
-                rowLoader.loadRows(mRows);
-            }
-        });
+                        // Section key pattern
+                        Pattern pattern = Pattern.compile("^homesection(\\d+)$");
 
+                        // Add sections to map first to make sure they stay in the correct order
+                        Map<Integer, HomeSectionType> sections = new TreeMap<>();
 
+                        for (String key : prefs.keySet()) {
+                            Matcher matcher = pattern.matcher(key);
+                            if (!matcher.matches()) continue;
 
-    }
+                            int index = Integer.parseInt(matcher.group(1));
+                            HomeSectionType sectionType = HomeSectionType.getByName(prefs.get(key));
 
-    private StdItemQuery getResumeQuery() {
-        StdItemQuery resumeItems = new StdItemQuery();
-        resumeItems.setMediaTypes(new String[] {"Video"});
-        resumeItems.setRecursive(true);
-        resumeItems.setImageTypeLimit(1);
-        resumeItems.setEnableTotalRecordCount(false);
-        resumeItems.setCollapseBoxSetItems(false);
-        resumeItems.setExcludeLocationTypes(new LocationType[] {LocationType.Virtual});
-        resumeItems.setLimit(50);
-        resumeItems.setFilters(new ItemFilter[]{ItemFilter.IsResumable});
-        resumeItems.setSortBy(new String[]{ItemSortBy.DatePlayed});
-        resumeItems.setSortOrder(SortOrder.Descending);
-        return resumeItems;
-    }
+                            if (sectionType != null)
+                                sections.put(index, sectionType);
+                        }
+
+                        // Fallback when no customization is done by the user
+                        rows.clear();
+
+                        if (sections.size() > 0) {
+                            for (HomeSectionType section : sections.values()) {
+                                addSection(section);
+                            }
+                        } else {
+                            for (HomeSectionType section : DEFAULT_SECTIONS) {
+                                addSection(section);
+                            }
+                        }
+
+                        loadRows();
+                    }
 
-    private void addLatestMovies() {
-        LatestItemsQuery latestMovies = new LatestItemsQuery();
-        latestMovies.setFields(new ItemFields[] {ItemFields.PrimaryImageAspectRatio, ItemFields.Overview});
-        latestMovies.setIncludeItemTypes(new String[]{"Movie"});
-        latestMovies.setImageTypeLimit(1);
-        latestMovies.setLimit(50);
-        mRows.add(new BrowseRowDef(mApplication.getString(R.string.lbl_latest_movies), latestMovies, new ChangeTriggerType[]{ChangeTriggerType.LibraryUpdated, ChangeTriggerType.MoviePlayback}));
+                    @Override
+                    public void onError(Exception exception) {
+                        exception.printStackTrace();
```

Comment:
The exception here should be logged using the application logger. `mApplication.getLogger()`

Reply:
Fixed